### PR TITLE
Fixes #30606 - replace mount_react_component in extlogout

### PR DESCRIPTION
**Pull request by MariaAga**
replace mount_react_component with react_component in external logout

### CODE REVIEW

File: app/helpers/external_logout_helper.rb [hunk 1]
```diff
@@ -11,12 +11,9 @@ def external_logout_props
...
-    content_tag(:div, nil, :class => container_class) +
-    mount_react_component(
+    content_tag(:div, react_component(
       'ExternalLogout',
-      ".#{container_class}",
-      external_logout_props.to_json,
-      flatten_data: true
-    )
+      external_logout_props
+    ), :class => container_class)
```
Comment (amirfefer) [inline app/helpers/external_logout_helper.rb]: Same thought from #7919

Comment (MariaAga) [reply app/helpers/external_logout_helper.rb]: Changed the css as well

**Comment by amirfefer** [comment]
[test katello]

File: app/helpers/external_logout_helper.rb [hunk 2]
```diff
@@ -10,13 +10,9 @@ def external_logout_props
...
   def mount_external_logout
-    container_class = "external_logout"
-    content_tag(:div, nil, :class => container_class) +
-    mount_react_component(
+    react_component(
       'ExternalLogout',
-      ".#{container_class}",
-      external_logout_props.to_json,
-      flatten_data: true
+      external_logout_props
```
Comment (Ron-Lavi) [inline app/helpers/external_logout_helper.rb]: Can transform into a one liner
```suggestion
      react_component('ExternalLogout', external_logout_props)
```

Comment (MariaAga) [reply app/helpers/external_logout_helper.rb]: awesome, thanks!

**Comment by ezr-ondrej** [comment]
We need rebase for tests to run.

**Review by ezr-ondrej — APPROVED** [review]
Great :+1: thanks @MariaAga !
And thanks @amirfefer and @laviro for the reviews :upside_down_face:

**Pull request merged**